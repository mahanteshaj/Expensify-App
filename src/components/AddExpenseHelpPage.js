import React from 'react'
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom'

const AddExpenseHelpPage = () => (
    <div>
        This is form AddExpenseHelpPage component
        </div>
)
export default AddExpenseHelpPage;