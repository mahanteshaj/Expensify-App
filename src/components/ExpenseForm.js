import React from 'react'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'

import { fail } from 'assert';
const now = moment()
console.log(now.format('MMM Do,YYYY'))
export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            error: '',
            description: props.expense ? props.expense.description : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            note: props.expense ? props.expense.note : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false

        }
    }
    onDescriptionChange = (e) => {
        console.log('sada')
        const description = e.target.value;
        this.setState(() => ({

            description: description
        }))

    }
    onAmountChange = (e) => {
        console.log('Amount has been changed!!!')
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }))
        }
     }
    onNoteChange = (e) => {
        console.log('note has been changed!!!')
        const note = e.target.value;
        this.setState(() => ({
            note

        }))
    }
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }))
        }
        //this.setState(()=>({createdAt}))

    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({
            calendarFocused: focused
        }))

    }
    onSubmit = (e) => {
        e.preventDefault()
        if (!this.state.description || !this.state.amount) {
            ///error message 
            this.setState(() => ({

                error: ' Please Provide description and amount'
            }))
        }
        else {
            error: '',
                this.props.onSubmit({
                    description: this.state.description,
                    amount: parseFloat(this.state.amount, 10) * 100,
                    createdAt: this.state.createdAt.valueOf(),
                    note: this.state.note
                })
            console.log('submitted')
        }
    }
    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <input
                        placeholder="description"
                        type="text"
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                        autoFocus
                    />
                    <input
                        placeholder="Amount"
                        type="text"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onFocusChange={this.onFocusChange}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                    <textarea
                        placeholder="Add a note for expense here..."
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    />
                    <button>AddExpense</button>
                </form>
                {!!this.state.error && <p>{this.state.error}</p>}

            </div>
        )
    }
}