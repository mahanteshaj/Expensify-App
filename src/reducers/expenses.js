import axios from 'axios'
const expensesReducerDefaultState = []

const expensesReducer = (state = expensesReducerDefaultState, action) => {

     
    switch (action.type) {
        case 'ADD_EXPENSE': console.log({...action.expense})
        fetch('http://localhost:8080/expensify', { method: 'POST' , headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        }, body: JSON.stringify(
        action.expense
        )
        })
            return [...state, action.expense];
        case 'REMOVE_EXPENSE':
            fetch('http://localhost:8080/expensify/'+action.expense.id,{method:'DELETE'})
            return state.filter(({ id }) => id !== action.expense.id)
        
        case 'EDIT_EXPENSE': fetch('http://localhost:8080/expensify/' + action.id, {
            method: 'PUT', headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }, body: JSON.stringify(
                action.updates
            )
        })
                
            return state.map((expense) => {
                if (expense.id == action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }

                }
                else {
                    return expense
                }
            })
         case 'SORT_BY_AMOUNT': 
                state.sort(function (action1, action2) {
            return action1.amount - action2.amount;

        })
        default:
            return state

    }
}
export default expensesReducer;