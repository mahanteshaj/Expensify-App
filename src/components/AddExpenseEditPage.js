import React from 'react'
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import ExepenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses';
export class AddExpenseEditPage extends React.Component {
    onSubmit = (expense) => {
        // console.log('updated', expense);
        // console.log('sdsd', this.props)
        this.props.editExpense(this.props.expense.id, expense)
        this.props.history.push('/')

    }
    onRemove = (expense) => {
        this.props.removeExpense({ id: this.props.expense.id })
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <ExepenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />

                <button onClick={this.onRemove}>Remove </button>

            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        editExpense: (id, expense) => {
            dispatch(editExpense(id, expense))
        },

        removeExpense: (id) => {
            dispatch(removeExpense(id))
        },
    }
}


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(AddExpenseEditPage);