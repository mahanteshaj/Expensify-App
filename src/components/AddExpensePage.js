import React from 'react'
import { connect } from 'react-redux'
import { addExpense } from '../actions/expenses'
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom'
import ExpenseForm from './ExpenseForm'

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        //  props.dispatch(addExpense(expense))
        this.props.addExpense(expense)
        this.props.history.push('/')
        console.log(expense)
    }
    render() {
        return (
            <div>
                This is form AddExpensePage component
            <h1>Expense Form</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )

    }


}
const mapDispatchToProps = (dispatch) => {
    return {
        addExpense: (expense) => {
            dispatch(addExpense(expense))
        }
    }
}

export default connect(undefined, mapDispatchToProps)(AddExpensePage);