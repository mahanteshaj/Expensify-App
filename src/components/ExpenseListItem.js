import React from 'react'
import {connect} from 'react-redux'
import { NavLink } from 'react-router-dom'
import {removeExpense} from '../actions/expenses'
export const ExpenseListItem=(props)=>(
    <div>
        <NavLink to={`/edit/${props.id}`}>
          <h3>{props.description}</h3>
        </NavLink>
       
    <p>{props.amount} -  {props.createdAt}
    </p> 
    </div>
)
export default connect()(ExpenseListItem)