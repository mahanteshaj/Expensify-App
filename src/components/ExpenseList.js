import React from 'react'
import { connect } from 'react-redux'
import filters from '../reducers/filters';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'
export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (<p>No expenses</p>) : (props.expenses.map((eachExpense) =>

                <ExpenseListItem key={eachExpense.id}{...eachExpense} />
            ))
        }
         </div>
)
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
        //filters:state.filters
    };
}
export default connect(mapStateToProps)(ExpenseList)