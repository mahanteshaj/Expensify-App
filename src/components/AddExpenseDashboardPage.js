import React from 'react'
import ExpenseList from './ExpenseList.js'
import ExpenseListItem from './ExpenseListItem.js';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom'
import ExpenseListFilters from './ExpenseListFilters'
export const ExpenseDashboardPage = () => (<div>
    This is from dashboard component
    <ExpenseListFilters/>
    <ExpenseList />
    <ExpenseListItem/>

</div>
)
export default ExpenseDashboardPage;