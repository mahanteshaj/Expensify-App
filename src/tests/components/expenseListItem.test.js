import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseListItem} from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

import {connect} from 'react-redux'
test('test should render with an expense list item',()=>{
  const   wrapper=shallow(<ExpenseListItem 
    { ...expenses[0]
    }/>)
  expect(wrapper).toMatchSnapshot();
})