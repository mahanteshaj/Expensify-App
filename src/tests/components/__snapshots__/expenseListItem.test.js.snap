// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should render with an expense list item 1`] = `
<div>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/edit/12223"
  >
    <h3>
      this  is description
    </h3>
  </NavLink>
  <p>
    1212
     -  
  </p>
</div>
`;
